aws-tickets.py

```diff
@@ -1,93 +1,100 @@
+                InstanceIds=[instance_tag]
             )
-        instance_status = ec2.describe_instance_status(
-          InstanceIds= [instance_tag]
-          )
-        for reservation in instance_name['Reservations']:
-          #print "Reservation %s " % reservation
-          for instance in reservation['Instances']:
-            for item in  instance['Tags']:
-              if item['Key'] == 'Name':
-                instance_details = {'Instance_Id': instance['InstanceId'], 'Name': item['Value']}
-                instance_event_dict.update({instance['InstanceId'] : instance_details})
-        for item in instance_status['InstanceStatuses']:
-          try:          
-            for event in item.get('Events',[]):
-              start_time = utc_to_local(event['NotBefore'])
-              end_time = utc_to_local(event['NotAfter'])
-              instance_event_dict[instance['InstanceId']].update({'Region' : region, 'Event_type' : event['Code'], 'Start': start_time, 'End': end_time})
-          except:
-            continue
-      except:
-        continue
-
-  return instance_event_dict
-
-def print_table(events):
-  x = PrettyTable(['Name', 'Instance ID', 'Region', 'Event_type', 'Start Local', 'End Local'])
-  for k, c in list(events.items()):
-    value = c.get("Region", "empty")
-    if value != "empty":
-      x.add_row([
-        c['Name'],
-        c['Instance_Id'],
-        c['Region'],
-        c['Event_type'],
-        c['Start'],
-        c['End']
-        ])
-  print(x)
-
-# def post_to_jira(environment, details)
-#   #jira_options={'server': 'http:/jira.clearslideng.com'}
-#   #jira=JIRA(options=jira_options,basic_auth=('user','password'))
-#   if environment == "dev":
-#     project = "SDLC"
-#   else:
-#     project = "OPS"
-#   issue_dict = {
-#     'project': {'key': project },
-#     'summary': 'New issue from jira-python',
-#     'description': 'Look into this one',
-#     'issuetype': {'name': 'Bug'},
-#      }
-#   new_issue = jira.create_issue(fields=issue_dict)
+            for reservation in instance_name['Reservations']:
+                # print "Reservation %s " % reservation
+                for instance in reservation['Instances']:
+                    for item in instance['Tags']:
+                        if item['Key'] == 'Name':
+                            instance_details = {'Instance_Id': instance['InstanceId'], 'Name': item['Value']}
+                            instance_event_dict.update({instance['InstanceId']: instance_details})
+
+            for item in instance_status['InstanceStatuses']:
+                for event in item['Events']:
+                    start_time = Utils.utc_to_local(event['NotBefore'])
+                    end_time = Utils.utc_to_local(event['NotAfter'])
+                    instance_event_dict[instance['InstanceId']].update(
+                        {'Event_type': event['Code'], 'Start': start_time, 'End': end_time})
+
+        return instance_event_dict
+
+
+class JIRA_API:
```

Comment:
thinking Eventual design is to create these in separate files IE Cloud, Tickets, Calendar. Not sure if that is excessive but it gives easier org.